lib/browser/app.js

```diff
@@ -59,6 +59,12 @@ app.controller('AppController', function($q, DriveScannerService, SelectionState
     this.selection.clear();
     this.writer.setProgress(0);
     this.scanner.start(2000);
+
+    this.scanner.scan().then(function (res) {
```

Comment:
I think this solution is fine except for non-critical issue: The drives are scanned twice at the same time when the app starts or get's restarted due to `scanner.start()` and `scanner.scan()`.

I have the following idea:

- Let's make `this.scanner.start()` accept a callback function that is called whenever the drives are scanned successfully, returning the array of detected drives. An alternative, which seems more "node-friendly" is to make `this.scanner.start()` return an `EventEmitter` instance that emits an event.

Then here, in the application controller, we can listen to this event and inject the appropriate logic.